design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,77 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+See: https://github.com/rook/rook/pull/10161
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low.
+
+## Proposed metrics for Rook to collect
+- `rook-version` - Rook version.
+  - Done in https://github.com/rook/rook/pull/10161
+- `rook-kubernetes-version` - Kubernetes version.
+  - This is similar to the already collected os/kernel versions.
+- `rook-num-kubernetes-nodes` - Number of Kubernetes nodes.
+- `rook-ceph-csi-version` - Ceph CSI version.
+- Num CSI plugin pods running - This gives idea of how many k8s nodes are in the k8s cluster that
+  are allowed to use Ceph storage.
+  - It is easiest count plugin pods separately since each DaemonSet is a different label. Plus, it
+    can show if some CSI drivers are ever disabled.
+  - `rook-num-csi-rbd-plugin-pods`
+  - `rook-num-csi-cephfs-plugin-pods`
+  - `rook-num-csi-nfs-plugin-pods`
+- `rook-num-nodes` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we should report an unknown value, possibly `unknown`.
+- `rook-num-storage-classes` - Number of storage classes related to the current Ceph cluster. We can
```

Comment:
Why do we need the aggregate number if we collect the individual counts?
```suggestion
- Number of storage classes related to the current Ceph cluster. We can
```

Reply:
I think we want to keep the overall count. This could be important if we add a new storage type in the future that isn't identified by a specific type.